pkg/services/ngalert/eval/eval.go

```diff
@@ -74,7 +75,21 @@ func (r *conditionEvaluator) EvaluateRaw(ctx context.Context, now time.Time) (re
 		execCtx = timeoutCtx
 	}
 	logger.FromContext(ctx).Debug("Executing pipeline", "commands", strings.Join(r.pipeline.GetCommandTypes(), ","), "datasources", strings.Join(r.pipeline.GetDatasourceTypes(), ","))
-	return r.expressionService.ExecutePipeline(execCtx, now, r.pipeline)
+	result, err := r.expressionService.ExecutePipeline(execCtx, now, r.pipeline)
+
+	// Check if the result of the condition evaluation is too large
+	if err == nil && result != nil && r.evalResultLimit > 0 {
```

Comment:
The interesting thing about putting the limit in EvaluateRaw is that it'll apply to Recording rules too. So, it's also a limit on the number of series a recording rule can generate.

I think that's okay, just as long as we're aware. The naming is generic enough that I think this doesn't warrant any changes.